include/max_element.hpp

```diff
@@ -21,11 +21,58 @@ namespace bit {
 
 
 
-// Status: to do
+// Status: complete
 template <class ForwardIt>
 constexpr bit_iterator<ForwardIt> max_element(bit_iterator<ForwardIt> first,
     bit_iterator<ForwardIt> last) {
-    (last);
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+    constexpr std::size_t num_digits = binary_digits<word_type>::value;
+
+    bit_iterator<ForwardIt> cursor = first;
+
+    // handle misaligned first word. for example ^00100^101
+    if (!_is_aligned_lsb(cursor)) {
+
+        // if 0 > 1 (version with comparators) we'll need to do this differently
+        // maybe use _padded_read?
+        word_type shifted_first_word = _shift_towards_lsb(*(cursor.base()), 
+            cursor.position());
+
+        if (shifted_first_word > 0) {
+            std::size_t first_set_bit_position = _tzcnt(shifted_first_word);
+            return cursor + first_set_bit_position;
+        } 
+
+        cursor = bit_iterator(std::next(cursor.base()));
+    }
+
+    ForwardIt word_cursor = cursor.base();
+
+    while (word_cursor != last.base()) {
+        word_type cur_word = *word_cursor;
+        if (cur_word > 0) {
+            std::size_t cur_set_bit_position = _tzcnt(cur_word);
+            return bit_iterator(word_cursor, cur_set_bit_position);
+        } 
+        ++word_cursor;
+    }
+
+    cursor = bit_iterator(word_cursor);
```

Comment:
I think we can avoid creating another bit iterator. The only remaining conditions depend on the alignment of the `last` iterator and the only time we use `cursor` is to access the base word anyways. Not worth holding up the PR, though.